File: anagram-detection.js
/* Anagram Detection
An anagram is the result of rearranging the letters of a word to produce a new word (see wikipedia).

Note: anagrams are case insensitive

Complete the function to return true if the two arguments given are anagrams of each other; return false otherwise.

Examples
"foefet" is an anagram of "toffee"

"Buckethead" is an anagram of "DeathCubeK"

STRINGS FUNDAMENTALS
*/

//comparing map way of solving it
var isAnagram = function(test, original) {

    if(test.length != original.length) return false;


    const lowerTest=test.toLowerCase()
    const lowerOriginal=original.toLowerCase()

    const testMap={}
    const originalMap={}
        
    for(let letter of lowerTest){
        if(!testMap.hasOwnProperty(letter)){
            testMap[letter]=1
        }else{
            testMap[letter]+=1
        }
    }

    for(let letter of lowerOriginal){
        if(!originalMap.hasOwnProperty(letter)){
            originalMap[letter]=1
        }else{
            originalMap[letter]+=1
        }
    }

    for(let letter in testMap){
        if(testMap[letter] != originalMap[letter]){
            return false
        }
    }

    return true
};


//Sort method of solving
// var isAnagram = function(test, original) {
//   const sortedTest=[...test.toLowerCase()].sort().join('')
//   const sortedOriginal=[...original.toLowerCase()].sort().join('')

//   return sortedTest == sortedOriginal
// };
  

File: python-solutions/anagram-detection.py
#Anagram Detection
# DESCRIPTION:
# An anagram is the result of rearranging the letters of a word to produce a new word (see wikipedia).

# Note: anagrams are case insensitive

# Complete the function to return true if the two arguments given are anagrams of each other; return false otherwise.

# Examples
# "foefet" is an anagram of "toffee"

# "Buckethead" is an anagram of "DeathCubeK"

# STRINGSFUNDAMENTALS

# write the function is_anagram
def is_anagram(test, original):
    
    if len(test) != len(original):
        return False
    
    l_test = sorted([l for l in test.lower()])
    l_original = sorted([l for l in original.lower()])
    print(l_test)
    print(l_original)
    
    if ''.join(l_test) == ''.join(l_original):
        return True
    else:
        return False
    